govcd/catalog_test.go

```diff
@@ -84,3 +96,179 @@ func (vcd *TestVCD) Test_DeleteCatalog(check *C) {
+	check.Assert(err, IsNil)
+
+	check.Assert(string(result), Matches, ".*Upload progress 100.00%")
+
+	//verify
+	catalog, err = org.FindCatalog(catalogName)
+	check.Assert(catalog.Catalog.CatalogItems[0].CatalogItem[0].Name, Equals, itemName)
+}
+
+func (vcd *TestVCD) Test_UploadOvf_error_withSameItem(check *C) {
+	catalogName := TestCreateCatalog + "5"
+	itemName := "item5"
+	setupCatalog(vcd, check, catalogName)
+
+	catalog, _ := findCatalog(vcd, check, catalogName)
+	AddToCleanupList(itemName, "catalogItem", vcd.org.Org.Name, "Test_UploadOvf")
+
+	//add item
+	uploadTask, err := catalog.UploadOvf(getCurrentPath()+OvaPath, itemName, "upload from test", 1024)
+	check.Assert(err, IsNil)
+	err = uploadTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	//execute
+	_, err = catalog.UploadOvf(getCurrentPath()+OvaPath, itemName, "upload from test", 1024)
+	check.Assert(err.Error(), Matches, ".*API Error: 500.*Catalog Item name is not unique within the catalog.*")
+}
+
+func (vcd *TestVCD) Test_UploadOvf_cleaned_extracted_files(check *C) {
+	catalogName := TestCreateCatalog + "6"
+	itemName := "item6"
+	setupCatalog(vcd, check, catalogName)
+
+	//check existing count of folders
+	oldFolderCount := countFolders()
+
+	catalog, _ := findCatalog(vcd, check, catalogName)
+	AddToCleanupList(itemName, "catalogItem", vcd.org.Org.Name, "Test_UploadOvf")
+
+	//execute
+	uploadTask, err := catalog.UploadOvf(getCurrentPath()+OvaPath, itemName, "upload from test", 1024)
+	check.Assert(err, IsNil)
+	err = uploadTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	//verify
+	check.Assert(oldFolderCount, Equals, countFolders())
+
+}
+
+func countFolders() int {
+	files, err := ioutil.ReadDir(os.TempDir())
+	if err != nil {
+		log.Fatal(err)
+	}
+	count := 0
+	for _, f := range files {
+		if strings.Contains(f.Name(), util.TmpDirPrefix+".*") {
+			count++
+		}
+	}
+	return count
+}
+
+func checkUploadOvf(vcd *TestVCD, check *C, ovaFileName, testCreateCatalog, itemName string) {
+	setupCatalog(vcd, check, testCreateCatalog)
+
+	catalog, org := findCatalog(vcd, check, testCreateCatalog)
+	AddToCleanupList(itemName, "catalogItem", vcd.org.Org.Name, "Test_UploadOvf")
+
+	//execute
+	uploadTask, err := catalog.UploadOvf(ovaFileName, itemName, "upload from test", 1024)
+	check.Assert(err, IsNil)
+	err = uploadTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	//verify
+	catalog, err = org.FindCatalog(testCreateCatalog)
+	check.Assert(catalog.Catalog.CatalogItems[0].CatalogItem[0].Name, Equals, itemName)
+}
+
+func findCatalog(vcd *TestVCD, check *C, testCreateCatalog string) (Catalog, AdminOrg) {
+	org := getOrg(vcd, check)
+	catalog, err := org.FindCatalog(testCreateCatalog)
+	check.Assert(err, IsNil)
+	return catalog, org
+}
+
+func setupCatalog(vcd *TestVCD, check *C, testCreateCatalog string) {
+	org := getOrg(vcd, check)
+
+	// creating catalog
+	adminCatalog, err := org.CreateCatalog(testCreateCatalog, TestCreateCatalogDesc, true)
```

Comment:
If this test is using an AdminCatalog, it should check whether `skipAdminTests` is set, and skip the test if it is.

Reply:
Will add.

Reply:
this isn't needed cause tests rung fine with org user or admin.